index.js

```diff
@@ -303,43 +324,86 @@ FastForEach.prototype.insertAllAfter = function(nodeOrNodeArrayToInsert, insertA
   return nodeOrNodeArrayToInsert;
 };
 
+// checks if the deleted data item should be handled with delay for a possible reuse at additions
+FastForEach.prototype.shouldDelayDeletion = function (data) {
+  return this.allowMoveNodes && (typeof data === "object" || typeof data === "function");
+}
+
+// get the cached index from the data item which points to its pendind deletion info
+FastForEach.prototype.getPenndingDeleteIndex = function (data) {
```

Comment:
I feel like this function should be inlined into `genPendingDeleteFor`.   Is there any reason it's broken out?

Incidentally, probably should be `getPen_dingDeleteIndex` :)